From 5bca8742e96bc5f3447be18e9c6ac5692cc3e3cd Mon Sep 17 00:00:00 2001
From: Mario Sanchez Prada <mario@endlessm.com>
Date: Thu, 30 Oct 2014 11:06:44 -0700
Subject: obex: Make sure obexd daemon is launched if needed when starting to
 watch it

This fixes a situation where files can't be sent to the computer (e.g.
from a phone) until you enable and then disable the bluetooth sharing
options at least once, as the shutting down will cause the obexd daemon
to run due to the UnregisterAgent D-Bus call in obex_agent_down().

https://bugzilla.gnome.org/show_bug.cgi?id=739417

diff --git a/src/obexpush.c b/src/obexpush.c
index 6424bd2..6d305d7 100644
--- a/src/obexpush.c
+++ b/src/obexpush.c
@@ -740,7 +740,7 @@ on_bus_acquired (GDBusConnection *connection,
 
 	self->obexd_watch_id = g_bus_watch_name_on_connection (self->connection,
 							       MANAGER_SERVICE,
-							       G_BUS_NAME_WATCHER_FLAGS_NONE,
+							       G_BUS_NAME_WATCHER_FLAGS_AUTO_START,
 							       obexd_appeared_cb,
 							       NULL,
 							       self,
-- 
cgit v0.10.1

